bundles/org.eclipse.ui.navigator/src/org/eclipse/ui/navigator/CommonNavigator.java

```diff
@@ -255,6 +266,72 @@ public void handleLifecycleEvent(SaveablesLifecycleEvent event) {
 			ColumnViewerToolTipSupport.enableFor(commonViewer);
 		}
 
+		// Immediate fallback: handle Ctrl+A at the Tree level
+		commonViewer.getTree().addKeyListener(new KeyAdapter() {
+			@Override
+			public void keyPressed(KeyEvent e) {
```

Comment:
Somehow it seems like a hack to handle this one key directly, especially given the key binding appears to be configurable:

<img width="831" height="327" alt="Image" src="https://github.com/user-attachments/assets/ffdf19bc-ebd4-40eb-afd5-d97a35e2852b" />

Reply:
Note, CommonViewer is a base class in CNF and can be extended. Adding key and part listeners that consumes key or register handlers here directly might affect others code. So please carefully change the code here, it is not only about Project Explorer.